### PR TITLE
doc: release-notes-4.0: add release notes for firmware subsystem

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
Includes notes on SCMI's introduction.

### CODE REVIEW

**Comment by LaurentiuM1234** [comment]
Wanted to add some notes on what the SCMI support includes. If people don't believe this is appropriate we can just close this and have it handled as part of https://github.com/zephyrproject-rtos/zephyr/pull/81236. The main idea is to have Zephyr users know that SCMI is now supported anyways

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -1193,6 +1193,14 @@ Libraries / Subsystems
 
     (:github:`79653`)
 
+* Firmware
+
+  * introduced basic support for ARM's System Control and Management Interface, which includes:
```
Comment (mmahadevan108) [inline doc/releases/release-notes-4.0.rst:1198]: Capital `I`

File: doc/releases/release-notes-4.0.rst [hunk 2]
```diff
@@ -1193,6 +1193,14 @@ Libraries / Subsystems
...
+
+    * subset of clock management protocol commands
```
Comment (mmahadevan108) [inline doc/releases/release-notes-4.0.rst:1200]: Capital `S` and same below.

**Review by kartben — DISMISSED** [review]

File: doc/releases/release-notes-4.0.rst (hunk 1 above)
Comment (LaurentiuM1234) [reply doc/releases/release-notes-4.0.rst]: fixed, thx

**Review by kartben — APPROVED** [review]

**Review by dkalowsk — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**